gigascience/gigadb-website
# add canonical URL to dataset pages

# Pull request for issue: #514

* Add canonical URL tag to dataset pages

## How to test?

* Navigate to any dataset URL, e.g. http://gigadb.gigasciencejournal.com/dataset/100006/
* Check that the head tag contains the same canonical URL
* Navigate to the paginated URL, eg. http://gigadb.gigasciencejournal.com/dataset/view/id/100006/Samples_page/1/Files_page/2 and test that the canonical did not change

## How have functionalities been implemented?

* Added a canonicalURL public variable to the DatasetController
* Define the value of canonicalURL right before the line that renders the public view
* In the dataset layout, render a `<link rel="canonical" href="<?= CHtml::encode($this->canonicalUrl) ?>" />` if the canonicalURL is defined

Questions:

* is it a good practice to define a new variable in the controller for this, or should it be done differently?

## Any issues with implementation?

--

## Any changes to automated tests?

--
